### PR TITLE
Lift ImportDepth utility out of SourceKit into IDE (NFC)

### PR DESCRIPTION
**Pull request by benlangmuir**

### CODE REVIEW

**Comment by benlangmuir** [comment]
@swift-ci please smoke test

File: lib/IDE/CodeCompletion.cpp [hunk 1]
```diff
@@ -7010,3 +7011,71 @@ void SimpleCachingCodeCompletionConsumer::handleResultsAndModules(
...
+
+      // Implicitly imported sub-modules get the same depth as their parent.
+      if (const clang::Module *CMI =
+              import.importedModule->findUnderlyingClangModule())
+        if (CM && CMI->isSubModuleOf(CM))
+          next = depth;
+      worklist.emplace_back(import.importedModule, next);
+    }
+  }
+}
```
Comment (rintaro) [inline lib/IDE/CodeCompletion.cpp:7081]: nitpick: `\n`

**Review by rintaro — APPROVED** [review]

**Comment by benlangmuir** [comment]
@swift-ci please smoke test

**Comment by benlangmuir** [comment]
@swift-ci please smoke test Linux

**Pull request merged**